string input
---
string output
bool success